homeassistant/components/venstar/sensor.py

```diff
@@ -155,9 +155,21 @@ def unique_id(self):
         return f"{self._config.entry_id}_{self.sensor_name.replace(' ', '_')}_{self.entity_description.key}"
 
     @property
-    def native_value(self) -> int:
+    def native_value(self) -> int | None:
         """Return state of the sensor."""
-        return self.entity_description.value_fn(self.coordinator, self.sensor_name)
+        try:
+            value = self.entity_description.value_fn(self.coordinator, self.sensor_name)
+            if self._attr_available and value is None:
+                raise ValueError
+            self._attr_available = True
+            return value
```

Comment:
While I agree that it should be done with the available property,  the only way to do that would be just implement another attribute that we touch here, or just check available based on the current value, i.e. return False if value is None else True, both would have the same problem.

That said,  my understanding is that because the entity is updated only by the update coordinator, and the update coordinator runs as long as one of the entities attached to it is available, the state is ultimately updated and therefor marked available when it comes back online.

This matches the experience shown in the graph above (https://github.com/home-assistant/core/pull/104954#issuecomment-1837579160)

Any preference for how to accomplish this?  or should I just remove availability for now, and add availability in another PR where discussions about the proper way to do it can be had?